packages/host/app/components/matrix/room.gts

```diff
@@ -151,6 +166,14 @@ export default class Room extends Component<Signature> {
     return this.cardsToSend.get(this.args.roomId);
   }
 
+  private get lastMessageSent() {
+    return this.lastMessagesSent.get(this.args.roomId);
+  }
+
+  private get isMessagePendingDisplayed() {
+    return (this.room?.messages.length ?? 0) === this.lastMessageSent?.index;
```

Comment:
Have you considered adding a client side id to the message payload when sending via matrix and then checking if a message with that id exists in room's messages? If it exists it means the message was sent otherwise it's pending. Perhaps It would help to get rid of a lot of logic added in this PR.

Reply:
Nice idea! I'll try to implement it.

Reply:
Updated [1203d7c](https://github.com/cardstack/boxel/pull/1107/commits/1203d7cb7c6028c7c1f452ce46e4b6108d62fad7)